crates/astria-sequencer-client/src/extension_trait.rs

```diff
@@ -648,21 +647,6 @@ pub trait SequencerClientExt: Client {
         Ok(native)
     }
 
-    /// Submits the given transaction to the Sequencer node.
-    ///
-    /// This method blocks until the transaction is checked, but not until it's committed.
-    /// It returns the results of `CheckTx`.
-    ///
-    /// # Errors
-    ///
-    /// - If calling the tendermint RPC endpoint fails.
-    async fn submit_transaction_sync(&self, tx: Transaction) -> Result<tx_sync::Response, Error> {
```

Comment:
This method is actually useful because here the caller need not know about how to transform a `Transaction` into a payload sequencer understands.

I would leave this.